libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -90,16 +108,13 @@ bool IRGeneratorForStatements::visit(VariableDeclarationStatement const& _varDec
 
 		expression->accept(*this);
 
-		solUnimplementedAssert(
-			*expression->annotation().type == *_varDeclStatement.declarations().front()->type(),
-			"Type conversion not yet implemented"
-		);
+		Type const& to = *_varDeclStatement.declarations().front()->type();
+
 		m_code <<
 			"let " <<
 			m_context.variableName(*_varDeclStatement.declarations().front()) <<
 			" := " <<
-			m_context.variable(*expression) <<
-			"\n";
+			convertIfRequired(*expression, to);
```

Comment:
What about "`expresionAsType`"? "convertIfRequired" does not really say what it converts.